### DISCUSSION TITLE
ipxe cosmetic customizations

### DISCUSSION THREAD

**Discussion by wfragakis** [ltsp/ltsp#332] [Q&A] [answered]
Hi all,
I've been experimenting with some tweaks to the standard ipxe configuration in ltsp.ipxe. One of the changes, btw, is a menu item to allow booting from the previous LTSP5 installation in case anything in our migration goes bad (missing features, etc). It's all been fairly straight forward after slogging through the ipxe docs, etc.

One thing I haven't been able to do, so far, is tweak the cosmetics of the ipxe menu, e.g. colors or background image. I'm wondering if the version included in LTSP is capable of it.

For example, if I want to use a splash image, the docs suggest that I can set one with
set console --picture {path to image} but using
set console --picture ltsp/ipxe.png or set console --picture tftp:/${srv}}/ltsp/ipxe.png 
neither seem to work.

Am I trying to do something beyond the capabilities of the included ipxe package or am I missing something in my syntax?

**Comment by alkisg** [comment]
ltsp-binaries is using the stock binaries from boot.ipxe.org.
When ltsp-binaries isn't installed, the Debian ipxe package contents are used instead, which again are normal ipxe images, not cut-down versions.

So everything that ipxe supports, should be supported. I haven't tested background images etc, you could test the dnsmasq logs to see if the client asks for missing files, but other than that, you should ask in #ipxe, which know their capabilities better.

That said, a note: VirtualBox contains an embedded ipxe ROM, which IS a cut-down version that only supports a few things. So if anyone's using VirtualBox, they should boot with ipxe.iso or ipxe.dsk instead of the embedded "boot from network". Or, they could install the extensions pack, when then uses a standard Intel UNDI ROM.

**Comment by wfragakis** [comment]
Many thanks. I will pursue your suggestions. As I mentioned, these are cosmetic tweaks so no urgency but your prompt reply is definitely appreciated

**Comment by maccadia** [comment, marked as answer]
Hello, I did some branding with ipxe. I use colours and picture :

```
colour --rgb 0x152a44 2
colour --rgb 0xffffff 7
cpair --foreground 2 --background 7 2

console  --x 1024 --y 768 --picture tftp://${net0/next-server}/ltsp/ipxe_uga.png --depth 32 --keep
```


I had to compile ipxe after changing some header files : 

1. uncomment `//#define CONSOLE_CMD` in src/config/general.h (not sure if this one is needed)
2. uncomment `//#define	CONSOLE_FRAMEBUFFER` in src/config/config.h
3. build ipxe with `make bin-x86_64-efi/snponly.efi bin/undionly.kpxe` to obtain snponly.efi (for x64 EFI) and undionly.kpxe (for legacy PXE)

Tell me if you need more details.

**Reply by wfragakis** [reply]
Thank you for the instructions. I presume the commented CONSOLE_etc were likely the source of my issues.

**Comment by maccadia** [comment]
I had some notes on iPXE we made with my colleague, I copied it on the LTSP wiki : https://github.com/ltsp/ltsp/wiki/Getting-started-with-iPXE
